packages/backend/src/api/job/jobModel.ts

```diff
@@ -0,0 +1,84 @@
+import { extendZodWithOpenApi } from '@asteasolutions/zod-to-openapi';
+import { commonValidations } from 'src/common/utils/commonValidation';
+import { z } from 'zod';
+
+extendZodWithOpenApi(z);
+
+export type Job = z.infer<typeof JobSchema>;
+
+export const JobSchema = z
+  .object({
+    id: z.string().uuid(),
+    title: z.string().min(2, 'Title cannot be empty').max(100, 'Title is too long'),
+    levelId: z.number().int().positive('Level ID must be a positive integer'),
+    categoryId: z.number().int().positive('Category ID must be a positive integer'),
+    typeId: z.number().int().positive('Type ID must be a positive integer'),
+    location: z.string().min(1, 'Location cannot be empty').max(100, 'Location is too long'),
+    description: z.string().min(10, 'Description is too short').max(5000, 'Description is too long'),
+    compensation: z.string().min(1, 'Compensation cannot be empty').max(100, 'Compensation description is too long'),
+    applicationLink: z.string().url('Application link must be a valid URL'),
+    isExternal: z.boolean(),
+    companyId: z.string().uuid().nullable(),
+    createdAt: z.union([z.string().datetime(), z.date()]),
+    updatedAt: z.union([z.string().datetime(), z.date()]),
+    postedAt: z.union([z.string().datetime(), z.date(), z.null()]),
+    tags: z.string().max(500, 'Tags are too long'),
+    isApproved: z.boolean(),
+    categoryTitle: z.string().optional(),
+    levelTitle: z.string().optional(),
+    companyName: z.string().nullable().optional(),
```

Comment:
Why do we have companyName ?

Reply:
I join the tables and return the the names in addition so the frontend can display the name directly, i tho this would be better but if u want I can remmove it